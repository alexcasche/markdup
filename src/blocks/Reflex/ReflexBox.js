import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { ReflexElement } from 'react-reflex'

const MODIFIERS = {
  expanded: () => `
    position: fixed !important;
    top: 0;
    left: 0;
    z-index: 999;
  `,
}

const ReflexBox = styled(ReflexElement)`
  overflow: hidden;
  @media only screen and (max-width: ${props =>
      props.theme.breakPoints.mobile}) {
    width: 100vw !important;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default ReflexBox
