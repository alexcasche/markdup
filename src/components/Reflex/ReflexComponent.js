import React, { Component } from 'react'
import { Reflex } from 'blocks'
import { Editor, Preview } from 'components'
import { PropTypes as t } from 'prop-types'

export default class ReflexComponent extends Component {
  constructor(props) {
    super(props)
    this.state = {
      expanded: false,
      mobilePreview: false,
      resizing: false,
    }
    this.resizeProps = {
      onResize: this.onResize,
      onStopResize: this.onStopResize,
    }
  }
  onResize = () => {
    this.setState(prevState => ({ ...prevState, resizing: true }))
  }
  onStopResize = () => {
    this.setState(prevState => ({ ...prevState, resizing: false }))
  }
  toggleExpand = box => {
    this.setState(prevState => ({
      ...prevState,
      expanded: !prevState.expanded ? box : false,
    }))
  }
  togglePreview = () => {
    this.setState(prevState => ({
      ...prevState,
      mobilePreview: !prevState.mobilePreview,
    }))
  }
  render() {
    const { expanded, mobilePreview, resizing } = this.state
    const { splitView } = this.props
    const containerModifiers = [
      ...(mobilePreview && !expanded ? ['mobilePreview'] : []),
    ]
    const boxModifiers = {
      preview: [...(expanded === 'preview' ? ['expanded'] : [])],
      editor1: [...(expanded === 'editor1' ? ['expanded'] : [])],
      editor2: [...(expanded === 'editor2' ? ['expanded'] : [])],
    }
    return (
      <Reflex
        className="Reflex"
        orientation="vertical"
        modifiers={containerModifiers}
      >
        <Reflex.Box minSize="250" {...this.resizeProps}>
          <Reflex orientation="horizontal">
            <Reflex.Box
              minSize="44"
              {...this.resizeProps}
              modifiers={boxModifiers.editor1}
            >
              <Editor
                events={{
                  toggleExpand: () => this.toggleExpand('editor1'),
                  togglePreview: this.togglePreview,
                }}
                expanded={expanded === 'editor1'}
                lang="markdown"
                resizing={resizing}
              />
            </Reflex.Box>
            {splitView ? <Reflex.Split /> : null}
            {splitView ? (
              <Reflex.Box
                minSize="46"
                {...this.resizeProps}
                modifiers={boxModifiers.editor2}
              >
                <Editor
                  events={{
                    toggleExpand: () => this.toggleExpand('editor2'),
                    togglePreview: this.togglePreview,
                  }}
                  expanded={expanded === 'editor2'}
                  lang="html"
                  resizing={resizing}
                />
              </Reflex.Box>
            ) : null}
          </Reflex>
        </Reflex.Box>
        <Reflex.Split />
        <Reflex.Box
          minSize="250"
          {...this.resizeProps}
          modifiers={boxModifiers.preview}
        >
          <Preview
            expanded={expanded === 'preview'}
            events={{
              toggleExpand: this.toggleExpand,
              togglePreview: this.togglePreview,
            }}
          />
        </Reflex.Box>
      </Reflex>
    )
  }
}

ReflexComponent.propTypes = {
  splitView: t.bool.isRequired,
}
