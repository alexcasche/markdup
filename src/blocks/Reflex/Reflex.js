import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { ReflexContainer } from 'react-reflex'
import { tint } from 'polished'
import 'react-reflex/styles.css'

import ReflexBox from './ReflexBox'
import { ReflexSplitter } from 'react-reflex'

const MODIFIERS = {
  mobilePreview: ({ theme }) => `
    @media only screen and (max-width: ${theme.breakPoints.mobile}) {
      transform: translateX(-50%);
    }
  `,
}

const Reflex = styled(ReflexContainer)`
  .reflex-splitter {
    z-index: 1 !important;
    border-color: ${props => tint(0.35, props.theme.colors.grey)} !important;
  }
  &.horizontal {
    .reflex-splitter {
      border-top: none !important;
      height: 1px !important;
    }
  }
  @media only screen and (max-width: ${props =>
      props.theme.breakPoints.mobile}) {
    &.vertical {
      width: 200vw !important;
    }
    .reflex-element {
      flex: 1 0 0% !important;
    }
    .reflex-splitter {
      display: none !important;
    }
  }
  ${applyStyleModifiers(MODIFIERS)};
`

Reflex.Box = ReflexBox
Reflex.Split = ReflexSplitter

export default Reflex
