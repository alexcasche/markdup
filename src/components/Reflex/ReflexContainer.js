import React from 'react'
import { connect } from 'react-redux'
import { getSettingVal } from 'selectors'
import ReflexComponent from './ReflexComponent'

export const ReflexContainer = props => {
  return <ReflexComponent {...props} />
}

function mapStateToProps(state) {
  return {
    splitView: getSettingVal(state, 'splitView'),
  }
}

export default connect(mapStateToProps)(ReflexContainer)
