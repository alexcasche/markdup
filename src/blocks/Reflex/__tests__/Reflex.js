import Reflex from '../Reflex'

describe('Reflex Block', () => {
  it('renders correctly', () => {
    const tree = render(<Reflex />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds mobilePreview modifier', () => {
    const tree = render(<Reflex modifiers={['mobilePreview']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('transform', 'translateX(-50%)', {
      media: 'only screen and (max-width:600px)',
    })
  })
})
