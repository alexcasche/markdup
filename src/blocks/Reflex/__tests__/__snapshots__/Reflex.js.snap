// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflex Block adds mobilePreview modifier 1`] = `
.c0 .reflex-splitter {
  z-index: 1 !important;
  border-color: #e1e2e4 !important;
}

.c0.horizontal .reflex-splitter {
  border-top: none !important;
  height: 1px !important;
}

@media only screen and (max-width:600px) {
  .c0.vertical {
    width: 200vw !important;
  }

  .c0 .reflex-element {
    -webkit-flex: 1 0 0% !important;
    -ms-flex: 1 0 0% !important;
    flex: 1 0 0% !important;
  }

  .c0 .reflex-splitter {
    display: none !important;
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

<div
  className="reflex-layout reflex-container horizontal c0"
  style={Object {}}
/>
`;

exports[`Reflex Block renders correctly 1`] = `
.c0 .reflex-splitter {
  z-index: 1 !important;
  border-color: #e1e2e4 !important;
}

.c0.horizontal .reflex-splitter {
  border-top: none !important;
  height: 1px !important;
}

@media only screen and (max-width:600px) {
  .c0.vertical {
    width: 200vw !important;
  }

  .c0 .reflex-element {
    -webkit-flex: 1 0 0% !important;
    -ms-flex: 1 0 0% !important;
    flex: 1 0 0% !important;
  }

  .c0 .reflex-splitter {
    display: none !important;
  }
}

<div
  className="reflex-layout reflex-container horizontal c0"
  style={Object {}}
/>
`;
